clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1036,7 +1041,8 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
   if (!actionRequiresCodeGen(Action) && CodeGenOpts.VerifyModule)
     MPM.addPass(VerifierPass());
 
-  if (Action == Backend_EmitBC || Action == Backend_EmitLL) {
+  if (Action == Backend_EmitBC || Action == Backend_EmitLL ||
+      CodeGenOpts.FatLTO) {
```

Comment:
What is the Action type with FatLTO?

Reply:
I believe its `Backend_EmitObj` .  I think that gets a little complicated when `-S` or `-emit-llvm` is passed, since I think we then select  `Backend_EmitLL` or `Backend_EmitBC`. https://github.com/llvm/llvm-project/blob/d82e93e7f129d9e8b72570efdf4a15d6ec3d4336/clang/lib/Driver/Driver.cpp#L4770

Reply:
I see, I guess then the check for the CodeGenOpt is needed here to catch this case.